apache/incubator-dolphinscheduler
# new feature for #404 add resource tree function

## What is the purpose of the pull request

*add resource tree function*
1.It supported directory mode,The resoruce files in diffrent directorys can has the same resource name.
2.It not support resource version.

## Brief change log

*add resource tree function*
  - *add resource filter and visitor in package org.apache.dolphinscheduler.api.dto.resources*
  - *add pid、is_directory and full_name in t_ds_resources*
  - *add abstract method setMainJarName in AbstractYarnTask*

## Verify this pull request
This change added tests and can be verified as follows:

*add the unit test*

  - *added ResourceFilterTest*
  - *update ResourceMapperTest*
  - *update ResourcesServiceTest*

the detail can also see https://github.com/apache/incubator-dolphinscheduler/issues/404
